Description: handle deprecation of String.create in transtion to OCaml 4.02
 String.create is deprecated in favor of Bytes.create in OCaml 4.02,
 causing a warning that is treated as a fatal error without this patch.
 But the Bytes module doesn't exist in 4.01, so we can't simply update the code
 until after the transition is complete.
 .
 approx (5.5-2) unstable; urgency=medium
 .
   * handle String.create deprecation warning during transition to OCaml 4.02
Author: Eric Cooper <ecc@cmu.edu>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- approx-5.5.orig/Makefile
+++ approx-5.5/Makefile
@@ -1,9 +1,10 @@
 # approx: proxy server for Debian archive files
-# Copyright (C) 2013  Eric C. Cooper <ecc@cmu.edu>
+# Copyright (C) 2015  Eric C. Cooper <ecc@cmu.edu>
 # Released under the GNU General Public License

 OCAMLBUILD := ocamlbuild
-OCAMLBUILD_OPTS := -classic-display
+# Don't fail on deprecation warning 3 during transition to OCaml 4.02
+OCAMLBUILD_OPTS := -classic-display -cflags -warn-error,A,-warn-error,-3

 TARGET := native

--- approx-5.5.orig/_tags
+++ approx-5.5/_tags
@@ -1,8 +1,9 @@
 # approx: proxy server for Debian archive files
-# Copyright (C) 2013  Eric C. Cooper <ecc@cmu.edu>
+# Copyright (C) 2015  Eric C. Cooper <ecc@cmu.edu>
 # Released under the GNU General Public License

-<**/*.{mli,ml}>: warn_error_A
+# Disable this during transition to OCaml 4.02 due to String.create deprecation
+#<**/*.{mli,ml}>: warn_error_A

 <approx.ml>: use_pcre, use_netstring, use_netcgi, use_nethttpd
 <config_file.ml>: use_pcre
